src/adv.py

```diff
@@ -66,34 +74,95 @@
+    cmd = input(
+        "What would you like to do? Select 'c' for a list of commands: ").lower()
 
-    if cmd in valid_cmds:
+    valid_cmds = ['s', 'n', 'e', 'w', 'q', 'c',
+                  'get', 'take', 'drop', 'i', 'inventory']
+    action = cmd.split()[0]
+
+    # if the first word entered is a valid command
+    if action in valid_cmds:
         direction = f"{cmd}_to"
         # grabs current_room of player, looks for direction attribute, defaults to None if invalid.
         new_room = getattr(player.current_room, direction, None)
+
+        # for directional commands n s e w
         if new_room:
             player.current_room = new_room
+        elif action == 'get' or action == 'take':
+            # if there is an item followed by the command check to see if the item is in the room
+            if len(cmd.split()) > 1:
+                requested_item = cmd.split()[1]
+                valid_items = getattr(
+                    player.current_room, "items")
+                grabbed_item = False
+                for i in valid_items:
+                    # if the requested item is in the room
+                    if i.name.lower() == requested_item:
+                        # add item to player inventory
+                        player.inventory.append(i)
+                        # remove item from current room
+                        player.current_room.items.remove(i)
+                        grabbed_item = True
+                        print("\n", i.on_take())
+                        break
+                if grabbed_item == False:
+                    print("That item is not in the room.")
+
+            else:
+                print("You need to include an item after your get command.")
+        elif action == 'drop':
+            if len(cmd.split()) > 1:
+                requested_item = cmd.split()[1].lower()
+                valid_items = player.inventory
+                dropped_item = False
+                for i in valid_items:
+                    # if the requested item is in player's inventory, drop it into the room
+                    if i.name.lower() == requested_item:
+                        player.current_room.items.append(i)
+                        player.inventory.remove(i)
+                        dropped_item = True
+                        print("\n", i.on_drop())
+                        break
+                if dropped_item == False:
+                    print("That item is not in your inventory!")
+            else:
+                print("You need to include an item after your get command.")
+        elif action == "i" or action == "inventory":
+            if len(player.inventory) < 1:
+                print("You have nothing in your inventory.")
+            else:
+                print("Your inventory contains the following:")
+                for i, item in enumerate(player.inventory, start=1):
+                    print(f"{i}) {item}")
```

Comment:
Good work with the main logic here. Remember we can always abstract repeated logic to functions.